src/Security/Authentication/Negotiate/src/NegotiateHandler.cs

```diff
@@ -328,6 +329,19 @@ protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
             {
                 Principal = user
             };
+
+            if (Options.LdapOptions.EnableLdapRoleClaimResolution)
```

Comment:
I'm realizing this option controls more than just the role claims. It controls if the LdapConnection is even created, the event, etc..
```suggestion
            if (Options.LdapOptions.EnableLdapClaimResolution)
```